Cython/Compiler/Nodes.py

```diff
@@ -2339,6 +2340,16 @@ def code_object(self):
         return self.py_func.code_object if self.py_func else None
 
     def analyse_declarations(self, env):
+        if self.decorators:
+            for decorator in self.decorators:
+                func = decorator.decorator
+                if func.is_name:
+                    if func.name == 'classmethod' or func.name == 'staticmethod':
+                        error(self.pos, "Cannot handle these decorators yet")
+                    if func.name == 'property':
+                        # XXX DO SOMETHING HERE???
+                        pass
+
```

Comment:
This XXX is hit if I remove `DecoratorTransform.visit_CFuncDefNode`, and the entry is registered properly for attribute lookup for `Foo.field0`, `Foo.field1`, `Foo.field2`, but then of course fails since the entry is a function `int (Foo)` not the function call.

Reply:
We'd need a pass, likely somewhere before AnalyzeTypes, that rewrites all Foo.field0 into Foo.field0().

Reply:
What is AnalyzeTypes? I turned on DebugFlags.debug_verbose_pipeline (since I am running a srctree test which runs cythonize in a subprocess, I needed to actually edit the file) and see a AnalyseDeclarationsTransform which is calling PropertyNode.analyse_declarations which is calling this function (CFuncDefNode.analyse_declarations) via line 5036 since it is part of the PropertyNode.body. But I couldn't figure out how to override the results of env.declare_property on line 5034 to make the CFuncDefNode's entry into the one called for `field0`

Reply:
Added a `ReplacePropertyNode` pass